### PR TITLE
M2 715 create age selector item

### PR DESCRIPTION
**Pull request by DimitrySakalouski**

### CODE REVIEW

File: src/shared/ui/survey/AgeSelector/AgeSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import Dropdown from '../../Dropdown';
+
+type DropdownOnChangeValue = {
+  value: string | number;
+};
+
+type Props = {
+  config: AgeSelectorConfigType;
+  value?: string | number;
+  onChange: (value: DropdownOnChangeValue) => void;
```
Comment (anq83) [inline src/shared/ui/survey/AgeSelector/AgeSelector.tsx:15]: DropdownOnChangeValue - maybe DropdownChangeValue or even DropdownValue, or SelectedAgeValue
Please think for better name

**Review by anq83 — CHANGES_REQUESTED** [review]

File: src/shared/ui/survey/AgeSelector/AgeSelector.tsx (hunk 1 above)
Comment (DimitrySakalouski) [reply src/shared/ui/survey/AgeSelector/AgeSelector.tsx]: Fixed

File: src/shared/ui/survey/AgeSelector/types.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+type AgeSelectorConfigType = {
```
[the remark this answers is not in the document]
Comment (DimitrySakalouski) [reply src/shared/ui/survey/AgeSelector/types.ts]: Fixed

File: src/shared/ui/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+const Dropdown: FC<DropdownProps> = ({
+  placeholder,
+  value = '',
+  onValueChange,
+  items,
+}) => {
+  const mappedItems = useMemo(
+    () => items.map(item => ({ label: item, value: item })),
```
Comment (BamMironov) [inline src/shared/ui/Dropdown.tsx:20]: Please define the following type:
```
type LabeledValue<TValue extends string | number = string> = {
  label: string;
  value: TValue;
};
```
`items` should already be of this type. So you don't have to map them here.
`value` should be `TValue`

`TValue` should come from `Dropdown` function

**Review by BamMironov — CHANGES_REQUESTED** [review]

**Review by anq83 — APPROVED** [review]

File: src/shared/ui/Dropdown.tsx (hunk 3 above)
Comment (DimitrySakalouski) [reply src/shared/ui/Dropdown.tsx]: Done

File: src/shared/ui/Dropdown.tsx [hunk 4]
```diff
@@ -16,13 +21,6 @@ const Dropdown: FC<DropdownProps> = ({
   onValueChange,
   items,
 }) => {
```
Comment (BamMironov) [inline src/shared/ui/Dropdown.tsx:23]:
```
function Dropdown<TValue extends string | number>({
  placeholder,
  value = '',
  onValueChange,
  items,
}: DropdownProps<TValue>) {
  ...
}
```

File: src/shared/ui/Dropdown.tsx [hunk 5]
```diff
@@ -1,12 +1,17 @@
...
+};
+
 type DropdownProps = {
   placeholder: string;
-  items: Array<number | string>;
-  value?: number | string;
+  items: LabeledValue[];
+  value?: LabeledValue;
   onValueChange: (value: number | string) => void;
 };
```
Comment (BamMironov) [inline src/shared/ui/Dropdown.tsx:16]:
```
type DropdownProps<TValue extends string | number> = {
  placeholder: string;
  items: LabeledValue<TValue>[];
  value?: LabeledValue<TValue>;
  onValueChange: (value: TValue) => void;
};
```

**Review by BamMironov — CHANGES_REQUESTED** [review]

**Review by BamMironov — APPROVED** [review]

**Pull request merged**